### ISSUE TITLE
Timestamp precision in EventSource input

### ISSUE THREAD

**Issue by ghelyar** [Azure/diagnostics-eventflow#134]
The [EventSource input](https://github.com/Azure/diagnostics-eventflow/blob/master/src/Microsoft.Diagnostics.EventFlow.Inputs.EventSource/EventDataExtensions.cs#L27) uses DateTime.UtcNow presumably because EventWrittenEventArgs does not contain a timestamp.

Would it be possible to use a more precise timestamp as described [here](https://msdn.microsoft.com/en-us/library/windows/desktop/dn553408%28v=vs.85%29.aspx)? The timestamps from the ETW input already use this level of precision, but of course they aren't added by EventFlow.

```
[DllImport("Kernel32.dll", CallingConvention = CallingConvention.Winapi)]
private static extern void GetSystemTimePreciseAsFileTime(out long filetime);

GetSystemTimePreciseAsFileTime(out long filetime);
var utcNow = DateTime.FromFileTimeUtc(filetime);
```

This requires interop and has minimum supported versions of Windows (8 and Server 2012), but perhaps it could be a configuration option or test whether it will work in a static constructor then fall back to DateTime.UtcNow if it won't, or something like that, or are these constraints too severe?

**Comment by ghelyar** [comment]
We're just trying to capture events as accurately as possible for analysis. DateTime.UtcNow is not really accurate enough to get decent durations out of start/stop events, for example. Some events appear too short or even instant, while others appear too long.

**Comment by karolz-ms** [comment]
Thanks, this helps. Would it then be sufficient if we used hi-res timer for just the full framework, and expect people to target Core 2.0 if they need hi-res timer for their Core apps?

**Comment by ghelyar** [comment]
I'm using net framework anyway because service fabric can't do net core yet (except for hosting it as a guest executable) so that's enough for me.

I would have thought that saying target net core 2.0 would be sufficient for net core users. There's probably no reason to implement it for < 2.0 unless someone asks for it and has a reason they can't target 2.0.

**Comment by karolz-ms** [comment]
Should be resolved with EventSource input version 1.1.3 or newer

Sorry it took so long to publish--we had major problems with our code signing infrastructure due to recent hurricanes
